dashboard/test/controllers/api/v1/sections_students_controller_test.rb

```diff
@@ -37,4 +37,27 @@ class Api::V1::SectionsStudentsControllerTest < ActionController::TestCase
     ].to_json
     assert_equal expected_summary, @response.body
   end
+
+  test 'teacher can update gender, name and age info for their student' do
+    sign_in @teacher
+    put :update, params: {section_id: @section.id, id: @student.id, gender: 'f', age: 9, name: 'testname'}
+    assert_response :success
+    assert_equal 'f', JSON.parse(@response.body)['gender']
+    assert_equal 9, JSON.parse(@response.body)['age']
+    assert_equal 'testname', JSON.parse(@response.body)['age']
+  end
+
+  test 'teacher can not update username info for their student' do
+    sign_in @teacher
+    current_username = @student.username
+    put :update, params: {section_id: @section.id, id: @student.id, username: 'newname'}
+    assert_response :success
+    assert_equal current_username, JSON.parse(@response.body)['username']
+  end
+
+  test 'non-owner can not update student info' do
+    sign_in @other_teacher
+    put :update, params: {section_id: @section.id, id: @student.id, gender: 'f'}
+    assert_response :forbidden
+  end
```

Comment:
I'd also add a test for the failure case. Send a bad gender value for instance, or stub `Student.any_instance.update` to raise an error, then `assert response :bad_request` and optionally verify the returned error json.